.travis.yml

```diff
@@ -0,0 +1,18 @@
+sudo: required
+language: node_js
+git:
+  depth: 1
+services:
+- docker
+before_install:
+- npm install
+- npm run lint
+install:
+- docker-compose up dev mongo-server selenium
+- docker-compose exec dev npm run deps
+- docker-compose exec dev npm start
+script:
+- docker-compose exec dev npm test
```

Comment:
few things:
we don't need webpack --watch, we just need to build it once
we don't need webpack or selenium at all for domain tests or REST API tests

Please try to follow this rule - execute test-actions one by one in order of "how much do I need to run it"
domain tests only need mongo-server and express to be running, and so are REST API tests
e2e tests require everything - selenium, mongo, express, webpack
So first start up mongo and express and run domain tests and, if possible, REST API tests
then build webpack (without --watch), then start up selenium, then run e2e tests

and comment out restriction "only master branch" for now, to test how it actually works on travis

Reply:
oh, and you forgot about `cp config.json.example config.json`

Reply:
actually, I think I was wrong - domain tests don't require express, only mongo, right?

Reply:
and if we specify memory db in config, they don't require even mongo, right?
so even better approach would be 
- run lint
- copy config, set memory db for tests in config
- run domain tests
- set mongo db for tests in config
- start mongo server
- run only db tests among domain tests
- webpack
- start selenium
- run e2e tests

Reply:
and don't do `npm run deps`, install node modules as they are required
before running domain tests, install node modules for domain,
before running express, install node modules for backend, etc

Reply:
we may even run memory domain tests before doing any docker-shenanigans